blockstore/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,382 @@
+        self.assertDictEqual(
+            response.data,
+            BundleVersionSerializer(
+                self.bundle_version, context={'request': response.wsgi_request}
+            ).data
+        )
+
+
+class CollectionViewSetTestCase(ViewsBaseTestCase):
+
+    def test_list(self):
+
+        url = reverse('api:v1:collection-list')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertListEqual(
+            response.data,
+            CollectionSerializer(
+                Collection.objects.all(), context={'request': response.wsgi_request}, many=True
+            ).data
+        )
+
+    def test_get(self):
+
+        url = reverse('api:v1:collection-detail', kwargs={'uuid': self.collection.uuid})
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertDictEqual(
+            response.data,
+            CollectionSerializer(
+                self.collection, context={'request': response.wsgi_request}
+            ).data
+        )
+
+    def test_create(self):
+
+        collections_count = Collection.objects.all().count()
+
+        url = reverse('api:v1:collection-list')
+        response = self.client.post(url, data={'title': 'Collection 2'}, format='json')
+
+        self.assertEqual(response.status_code, 201)
+        self.assertDictEqual(
+            response.data,
+            CollectionSerializer(
+                Collection.objects.all().order_by('-id').first(), context={'request': response.wsgi_request}
+            ).data
+        )
+        self.assertEqual(Collection.objects.all().count(), collections_count + 1)
+
+    def test_update(self):
+
+        collections_count = Collection.objects.all().count()
+
+        url = reverse('api:v1:collection-detail', kwargs={'uuid': self.collection.uuid})
+        response = self.client.patch(url, data={'title': 'Collection 1.1'}, format='json')
+
+        self.assertEqual(response.status_code, 200)
+        self.assertDictEqual(
+            response.data,
+            CollectionSerializer(
+                Collection.objects.get(uuid=self.collection.uuid), context={'request': response.wsgi_request}
+            ).data
+        )
+        self.assertEqual(Collection.objects.all().count(), collections_count)
+
+
+class BundleFileReadOnlyViewSetTestCase(ViewsBaseTestCase):
+
+    def test_list(self):
+
+        url = reverse('api:v1:bundleversionfile-list', kwargs={
+            'bundle_uuid': self.bundle_version.bundle.uuid, 'version_num': self.bundle_version.version_num
+        })
+        response = self.client.get(url)
+
+        # response.wsgi_request is the Django Request object and not the DRF Request object.
+        response.wsgi_request.parser_context = {
+            'kwargs': {
+                'bundle_uuid': self.bundle_version.bundle.uuid,
+                'version_num': self.bundle_version.version_num,
```

Comment:
Nit: When your test cases have to know this muchabout the internals of the serializers etc. (the above few lines as well as things like `'detail_view_name': BundleFileReadOnlyViewSet.detail_view_name`), it might be better to just check the output directly, instead of using the serializer to generate the expected output?

e.g.
```
response = self.client.get(url)
self.assertEqual(
    response.data,
    [
        {
            "path": 'a/file-1.html'
            "public": False,
            "size": 28,
            "url": response.data[0].url,
            "data": response.data[0].data,
        },
        {
            "path": 'b/file-2.txt'
            "public": False,
            "size": 20,
            "url": response.data[1].url,
            "data": response.data[1].data,
        },
    ],
)
```

I also find that much easier to read and understand.

Reply:
Hmm. Interesting points.

The reason for adding this is that DRF creates a wrapper object around the Django `Request` object which has all this data but for some reason it is not setting it on the API client response so we cannot use it. One option would be to add it ourselves but I just decided to go with this because of time constraints. Can look at this in the next round.

As for checking the output directly, it depends on whether we want to check all the details that have been tested in the serializer test cases as well? These tests are only running through the happy path right now, so we can update them in the next round when we add more.

Reply:
Added a point about this to LX-53.

Reply:
> As for checking the output directly, it depends on whether we want to check all the details that have been tested in the serializer test cases as well?

Personally, I do believe that it's better to err on the side of simple, readable, understandable test cases, even when that means testing multiple things at once (and having redundant tests), compared to test cases with lots of mocking and extra helper code added to ensure that they are only testing one thing at a time, which usually end up being harder to read and understand. But I recognize that both are valid approaches and it's fine to leave this as-is.

Reply:
Sure, it is possible this makes more sense for these tests. There were already a number of things to setup from scratch in this round so would be happy to think more deeply about it in the next round.